db/migrate/20190327173430_create_items.rb

```diff
@@ -0,0 +1,13 @@
+class CreateItems < ActiveRecord::Migration[5.2]
+  def change
+    create_table :items do |t|
+      t.string :name
+      t.string :description
+      t.float :price
+      t.references :bill , foreign_key: true
+      #Needed for nested attributes 
+      t.references :item, index: true, foreign_key: true
```

Comment:
What is this needed to do? Is this something that you could otherwise accomplish with a self-join instead?

https://guides.rubyonrails.org/association_basics.html#self-joins

Reply:
Cami wanted it added for nested forms, she used information from https://medium.com/@oliviazyc/complex-nesting-forms-in-rails-631b369be0de to learn about it.

Reply:
Thanks for sharing that link! The part that doesn't make sense to me here, though, is that you've nested a foreign key for `items` inside of the `items` table itself, which I don't think will work.